numpy/core/src/multiarray/compiled_base.c

```diff
@@ -80,6 +80,374 @@ minmax(const npy_intp *data, npy_intp data_len, npy_intp *mn, npy_intp *mx)
     *mx = max;
 }
 
+NPY_NO_EXPORT PyObject *
+arr_histogram_uniform(PyObject *NPY_UNUSED(self), PyObject *args, PyObject *kwds)
+{
+    PyObject *obj_x, *obj_edges, *weights = Py_None;
+    PyArrayObject *arr_x = NULL, *arr_edges = NULL, *arr_w = NULL, *hist = NULL;
+    npy_intp len, bins;
+    double *e, min, max, norm;
+
+    static char *kwlist[] = {"x", "edges", "weights", NULL};
+
+    if (!PyArg_ParseTupleAndKeywords(args, kwds, "OO|O:_histogram_uniform",
+            kwlist, &obj_x, &obj_edges, &weights)) {
+        goto fail;
+    }
+
+    arr_x = (PyArrayObject *)PyArray_ContiguousFromAny(obj_x, NPY_DOUBLE, 1, 1);
+    if (arr_x == NULL) {
+        goto fail;
+    }
+    len = (npy_intp)PyArray_DIM(arr_x, 0);
+
+    arr_edges = (PyArrayObject *)PyArray_ContiguousFromAny(obj_edges, NPY_DOUBLE, 1, 1);
+    if (arr_edges == NULL) {
+        goto fail;
+    }
+    bins = (npy_intp)(PyArray_DIM(arr_edges, 0) - 1);
+
+    /* Handle empty array */
+    if (len == 0) {
+        hist = (PyArrayObject *)PyArray_ZEROS(1, &bins, NPY_INTP, 0);
+        if (hist == NULL) {
+            goto fail;
+        }
+        Py_DECREF(arr_edges);
+        Py_DECREF(arr_x);
+        return (PyObject *)hist;
+    }
+
+    e = (double *)PyArray_DATA(arr_edges);
+
+    min = e[0];
+    max = e[bins];
+
+    if (bins <= 0)
+    {
+        PyErr_SetString(PyExc_ValueError, "Must supply at least one bin (two edges)");
+        goto fail;
+    }
+    if ((max - min) <= 0)
+    {
+        PyErr_SetString(PyExc_ValueError, "Bin edges must be increasing");
```

Comment:
I think this might be a regression

Reply:
how do you mean. Just overzealous error checking? Should I remove this?